azure-iot-device/azure/iot/device/common/alarm.py

```diff
@@ -0,0 +1,36 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+from threading import Thread, Event
+import time
+
+
+class Alarm(Thread):
+    """Call a function at a specified time"""
+
+    def __init__(self, alarm_time, function, args=None, kwargs=None):
+        Thread.__init__(self)
+        self.alarm_time = alarm_time
+        self.function = function
+        self.args = args if args is not None else []
+        self.kwargs = kwargs if kwargs is not None else {}
+        self.finished = Event()
+
+    def cancel(self):
+        """Stop the alarm if it hasn't finished yet."""
+        self.finished.set()
+
+    def run(self):
```

Comment:
>run [](start = 8, length = 3)

can u say what it does like if the current time is greater then something otherwise something else

Reply:
If the alarm has already expired, it immediately resolves the associated function. I will add a test to show this.